nemo_aligner/algorithms/spin.py

```diff
@@ -127,6 +128,11 @@ def __init__(
         self.length_params = OmegaConf.to_container(self.model.cfg.spin.length_params, resolve=True)
         self.sampling_params = OmegaConf.to_container(self.model.cfg.spin.sampling_params, resolve=True)
         self.max_gen_seq_len = self.length_params["max_length"]
+        dp_batch_size = self.model.cfg.global_batch_size // parallel_state.get_data_parallel_world_size()
+        assert (
+            self.model.cfg.spin.rollout_micro_batch_size % dp_batch_size == 0
+        ), f"rollout_micro_batch_size [{self.model.cfg.spin.rollout_micro_batch_size}] must be a multiple of GBS [{self.model.cfg.global_batch_size}] // DP [{parallel_state.get_data_parallel_world_size()}]"
+        self.rollout_micro_batch_size = max(self.model.cfg.spin.rollout_micro_batch_size, dp_batch_size)
```

Comment:
The max should always be `self.model.cfg.spin.rollout_micro_batch_size` (unless accidentally set to 0), right?
```suggestion
        self.rollout_micro_batch_size = self.model.cfg.spin.rollout_micro_batch_size
        assert self.rollout_micro_batch_size > 0, "`rollout_micro_batch_size` must be > 0"
```

Reply:
No, because as the code exists right now in this PR (which I would like to merge), you cannot have rollout_MBS < dp_batch_size, as that would lead to a crash. That's why this max statement is here, to ensure that rollout_MBS is always equal to at least the DP size.

Reply:
Not sure I follow, you have
```python
 assert self.model.cfg.spin.rollout_micro_batch_size % dp_batch_size == 0
```
which means rollout_MBS is a multiple of dp_batch_size, so as long as it's non-zero it has to be larger.

Reply:
Agreed, but what if rollout_MBS is zero? Instead of an assert, I have the max there to just increase it to DP size.

Reply:
Okay, I did it your way, resolving